src/ompl/geometric/planners/rrt/RRTstar.h

```diff
@@ -148,42 +160,125 @@ namespace ompl
+                As the ancestory of the descendent vertices can change at anytime, this means that the removed
+                descendents may actually be capable of later providing a better solution once
+                their incoming path passes through a different vertex (e.g., a change in homotopy class) */
+            void setTreePruning(const bool prune)
             {
-                prune_ = prune;
+                useTreePruning_ = prune;
             }
 
             /** \brief Get the state of the pruning option. */
-            bool getPrune() const
+            bool getTreePruning() const
             {
-                return prune_;
+                return useTreePruning_;
             }
 
-            /** \brief Set the percentage threshold (between 0 and 1) for pruning the tree. If the new tree has removed
-                at least this percentage of states, the tree will be finally pruned. */
-            void setPruneStatesImprovementThreshold(const double pp)
+            /** \brief Set the fractional change in solution cost necessary for pruning to occur, i.e.,
+                prune if the new solution is at least X% better than the old solution.
+                (e.g., 0.0 will prune after every new solution, while 1.0 will never prune.) */
+            void setPruneThreshold(const double pp)
             {
-                pruneStatesThreshold_ = pp;
+                pruneThreshold_ = pp;
             }
 
             /** \brief Get the current prune states percentage threshold parameter. */
-            double getPruneStatesImprovementThreshold () const
+            double getPruneThreshold() const
+            {
+                return pruneThreshold_;
+            }
+
+            /** \brief Controls whether heuristic rejection is used on samples (e.g., x_rand) */
+            void setSampleRejection(const bool reject);
+
+            /** \brief Get the state of the sample rejection option */
+            bool getSampleRejection() const
+            {
+                return useRejectionSampling_;
+            }
+
+            /** \brief Controls where heuristic rejection is used on new states before connection (e.g., x_new = steer(x_nearest, x_rand)) */
+            void setNewStateRejection(const bool reject)
+            {
+                useNewStateRejection_ = reject;
+            }
+
+            /** \brief Get the state of the new-state rejection option */
+            bool getNewStateRejection() const
+            {
+                return useNewStateRejection_;
+            }
+
+            /** \brief Controls search focusing. Search focusing consists of pruning the existing search and limiting future search.
+            In RRT* this is accomplished by turning on Informed RRT* (tree pruning and focused sampling) and new-state rejection.
+            This flag individually sets the options described above.
+            */
+            void setFocusSearch(const bool focus)
```

Comment:
This is the "meta" parameter to enable the generalized concept of focused search to facilitate use with CFOREST. Apparently I was less humble here and decided that the sample focusing would come from the informed sampling instead of rejection sampling.